app/src/main/java/com/codepath/michfeng/songswiper/LoginActivity.java

```diff
@@ -1,19 +1,84 @@
 package com.codepath.michfeng.songswiper;
-
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.content.Intent;
+import android.content.SharedPreferences;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.Window;
+
+import com.android.volley.RequestQueue;
+import com.android.volley.toolbox.Volley;
+import com.spotify.sdk.android.auth.AuthorizationClient;
+import com.spotify.sdk.android.auth.AuthorizationRequest;
+import com.spotify.sdk.android.auth.AuthorizationResponse;
 
 import java.net.Authenticator;
 
 public class LoginActivity extends AppCompatActivity {
 
+    public static final String TAG = "LoginActivity";
+    private static final String CLIENT_ID = "7d2e57d7cf85444da0db0412f4116c80";
+    private static final String REDIRECT_URI = "com.spotify.michfeng.songswiper://callback";
+    private static final int REQUEST_CODE = 1337; // what does this mean? ***
+    private static final String SCOPES = "user-read-recently-played,playlist-read-private,user-modify-playback-state" +
```

Comment:
Can you please add a comment in the code that lists the scopes and describes them?